software/control_sw/src/cosmic_fengine.py

```diff
@@ -879,14 +915,177 @@ def set_delay_tracking(self, delays, delay_rates, phases, phase_rates, load_time
             self.phaserotate.set_delay_rate(i, delay_rate)
             self.phaserotate.set_phase(i, phases[i])
             self.phaserotate.set_phase_rate(i, phase_rates_per_spec[i])
+    
+    def stop_delay_tracking(self):
+        if not self.delay_tracking_thread.is_alive():
+            self.logger.error(f"Delay tracking thread is stopped. Ignoring request.")
+            return 
+        self.delay_switch.clear()
+        self.delay_tracking_thread.join()
+
+        #Unsubscribe to the required redis channels
+        if self.redis_obj is not None:
+            for channel in self.delay_channel_names:
+                try:
+                    self.pubsub.unsubscribe(channel) 
+                except redis.RedisError:
+                    self.logger.warn(f'Unsubscription from `{channel}` unsuccessful.')    
+        else:
+            self.logger.warn("""Cosmic Fengine has no redis object instance. No subscribed channels
+            to unsubscribe from.""")
+            return
+        self.delay.initialize()
+        self.phaserotate.initialize()
+
+    def start_delay_tracking(self):
+        if self.delay_tracking_thread.is_alive():
+            self.logger.error(f"Delay tracking thread is running. Ignoring request.")
+            return
+        self.delay_switch.set()
+
+        #Subscribe to the required redis channels
+        if self.redis_obj is not None:
+            self.pubsub = self.redis_obj.pubsub(ignore_subscribe_messages=True)
+            for channel in self.delay_channel_names:
+                try:
+                    self.pubsub.subscribe(channel) 
+                except redis.RedisError:
+                    self.logger.error(f'Subscription to `{channel}` unsuccessful.')
+        else:
+            self.logger.warn("""Cosmic Fengine has no redis object instance. Cannot load delays
+            for delay tracking.""")
+            return
+
+        self.delay_tracking_thread = threading.Thread(
+            target=self.delay_tracking, args=(), daemon=False
+        )
+        self.delay_tracking_thread.start()
+
+    def delay_tracking(self):
+        """
+        Started in a thread, this function calls on delay_tracking to update the fengine coefficients for 
```

Comment:
"This function calls on delay_tracking": What does this mean, this function _is_ `delay_tracking`?

Reply:
Fixed in this [commit ](https://github.com/realtimeradio/vla-dev/commit/c3a6998176d9f20190d81a8da7b6821c66400913)